core/lib/i18n/i18n.js

```diff
@@ -189,8 +189,27 @@ function createIcuMessageFn(filename, fileStrings) {
     const keyname = Object.keys(mergedStrings).find(key => mergedStrings[key] === message);
     if (!keyname) throw new Error(`Could not locate: ${message}`);
 
-    const filenameToLookup = keyname in fileStrings ? filename : getModulePath(import.meta);
-    const unixStyleFilename = path.relative(LH_ROOT, filenameToLookup).replace(/\\/g, '/');
+    let filenameToLookup;
+    if (path.isAbsolute(filename)) {
+      // `message` can be a UIString defined within the provided `fileStrings`, or it could be
+      // one of the common strings found in `i18n.UIStrings`.
+      if (keyname in fileStrings) {
+        filenameToLookup = filename;
+      } else if (keyname in UIStrings) {
+        filenameToLookup = getModulePath(import.meta);
+      } else {
+        throw new Error(`Provided UIString is invalid: ${filename}, ${keyname}.`);
+      }
+
+      filenameToLookup = path.relative(LH_ROOT, filenameToLookup);
+    } else {
+      // `filename` might have been passed in as the exact i18n identifier
+      // already (see: stack-packs.js). Also the case for bundled lighthouse.
```

Comment:
> Also the case for bundled lighthouse

this case still needs the `UIStrings` fallback for the bundled case, though. The `en-US` fallback will still work because that was passed in, but e.g. that FCP example you end up with an `i18nId` of `'core/audits/metrics/first-contentful-paint.js | firstContentfulPaintMetric'`, not `'core/lib/i18n/i18n.js | firstContentfulPaintMetric'` like it should be, so it can't be localized.

What about adding 
`if (path.isAbsolute(filename)) filename = path.relative(LH_ROOT, filename);`

up at the top of the function below the somewhat similar `'file://'` correction? That should cover all these cases, at least

and that should cover everything?